pkg/kubelet/dockershim/docker_image.go

```diff
@@ -120,7 +120,7 @@ func (ds *dockerService) RemoveImage(_ context.Context, r *runtimeapi.RemoveImag
 	// TODO: We assume image.Image is image ID here, which is true in the current implementation
 	// of kubelet, but we should still clarify this in CRI.
 	imageInspect, err := ds.client.InspectImageByID(image.Image)
-	if err == nil && imageInspect != nil && len(imageInspect.RepoTags) > 1 {
+	if err == nil {
```

Comment:
Before that , kubelet remove it according to imageID , now we remove it to according to imageTag, does it work for you special condition ? and why ?